google_monitoring/uptime_checks.tf

```diff
@@ -0,0 +1,24 @@
+resource "google_monitoring_uptime_check_config" "https" {
+  for_each = { for uptime_check in var.uptime_checks : uptime_check.name => uptime_check }
```

Comment:
I'm not convinced it's useful to have this loop inside the module, rather than using
```
module "uptime_check" {
    source = "..."
    for_each = {...}
    ...
}
```
when instantiating the module and you want more than one check. I'd personally prefer to do loops when instatntiating the module. I think that's more flexible, since it makes it easier to share, say, the `user_labels` across all your checks.

(It probably doesn't matter much either way.)

Reply:
Thinking about this from the `symbols` project perspective, where we have two services sharing the same GCP project, would this module-level approach prevent us from having different user labels for the different services?

Reply:
Either approach would allow having different labels for different checks. As I said, it probably doesn't matter much either way. I just think it's a bit weird to have that loop inside the module. We don't do that for the database modules, so why should we do it here? If someone wants a loop, they can always add to the instantiation.

Reply:
fair point, this is currently based on the idea of the module calling project passing the checks in as a list, module takes care of working the details out .. I'm thinking about passing this in from e.g. a yaml file or something like this (that's easy to read and update by clients), working with it can still be done via locals easily .. removing the loop from the module might make it less complex, and easier for the users to maintain :thinking:

Reply:
I love the flexibility of the `optional` keyword in objects a lot, so personally always going for that .. also meant this to eventually become something that could also include other things like uptime_checks, so wanted to have a single list / map referring to the uptime checks that should be created :thinking:

Reply:
if it would remain to be only creating uptime_checks, I could name it differently and make the uptime_check relevant vars top level vars :thinking: